enterprise/backend/src/metabase_enterprise/gsheets/api.clj

```diff
@@ -180,10 +180,21 @@
     (throw (ex-info "Cannot fetch Google Drive connection: ID is nil" {})))
   (hm.client/make-request :get (str "/api/v2/mb/connections/" id)))
 
+(defn- url-type [url]
+  (cond
+    (re-matches #"^https://docs.google.com/spreadsheets/.*" url)
+    "google_spreadsheet"
+
+    (re-matches #"^https://drive.google.com/drive/.*" url)
+    "gdrive"
```

Comment:
I'd reccomend losening these a bit, in light of the frontend validation (which allows http or https prefixes). Not sure which way is 'right', but they should match.

I see the purpose of `url-type`, and we never had backend validation for this, so I think just making sure we can figure out the type here is better than validating it so strictly.

```suggestion
    (re-matches #".*docs.google.com/spreadsheets/.*" url)
    "google_spreadsheet"

    (re-matches #".*drive.google.com/drive/.*" url)
    "gdrive"
```